pex/common.py

```diff
@@ -363,7 +400,17 @@ def __str__(self):
   def delete(self):
     shutil.rmtree(self.chroot)
 
-  def zip(self, filename, mode='w'):
+  def zip(self, filename, mode='w', deterministic_timestamp=False):
     with open_zip(filename, mode) as zf:
       for f in sorted(self.files()):
-        zf.write(os.path.join(self.chroot, f), arcname=f, compress_type=zipfile.ZIP_DEFLATED)
+        full_path = os.path.join(self.chroot, f)
+        zinfo = zf.zip_info_from_file(
+            filename=full_path,
+            arcname=f,
+            date_time=(
+              None if not deterministic_timestamp else DETERMINISTIC_DATETIME.timetuple()[:6]
```

Comment:
This is the last wonky bit I can see. The `zip_info_from_file` already knows [:6] magic, on the other hand - the extracted constant is not in final form. I think probably having `zip_info_from_file` take a `time.struct_time` and storing what is currently `DETERMINISTIC_DATETIME.timetuple()` as `DETERMINISTIC_DATETIME` would clean this up.